src/extra/gpkg.jl

```diff
@@ -0,0 +1,1088 @@
+    min_z, max_z, min_m, max_m
+  )
+  
+  return header, position(io)
+end
+
+function read_coordinate_values(io::IOBuffer, byte_order::UInt8, has_z::Bool, has_m::Bool)
+  read_value = byte_order == 0x01 ? ltoh : ntoh
+  
+  x = read_value(read(io, Float64))
+  y = read_value(read(io, Float64))
+  
+  coords_list = [x, y]
+  
+  if has_z
+    z = read_value(read(io, Float64))
+    push!(coords_list, z)
+  end
+  
+  if has_m
+    m = read_value(read(io, Float64))
+  end
+  
+  return coords_list
+end
+
+function create_point_from_coords(coords_list::Vector{Float64}, crs_type)
+  x, y = coords_list[1], coords_list[2]
+  
+  if isnan(x) && isnan(y)
+    return nothing
+  end
+  
+  # Handle different CRS types properly
+  if isnothing(crs_type)
+    # No CRS specified - use raw coordinates as Cartesian
+    return Point(coords_list...)
+  end
+  
+  # Determine CRS characteristics using CoordRefSystems.jl
+  try
+    # For geographic coordinate systems (LatLon family)
+    if crs_type <: LatLon
+      # WKB stores geographic coordinates as x=longitude, y=latitude
+      # LatLon constructor expects (latitude, longitude) order
+      if length(coords_list) == 2
+        return Point(crs_type(y, x))  # LatLon(lat, lon) from WKB(x=lon, y=lat)
+      else
+        # 3D geographic with altitude/elevation
+        return Point(crs_type(y, x, coords_list[3]))
+      end
+    elseif crs_type <: LatLonAlt
+      # LatLonAlt always expects (lat, lon, alt)
+      if length(coords_list) >= 3
+        return Point(crs_type(y, x, coords_list[3]))
+      else
+        # Fallback to 2D LatLon if no Z coordinate
+        return Point(LatLon{datum(crs_type)}(y, x))
+      end
+    else
+      # For projected coordinate systems (Cartesian, UTM, etc.)
+      # WKB coordinate order matches CRS constructor order (x, y, [z])
+      if length(coords_list) == 2
+        return Point(crs_type(x, y))
+      else
+        return Point(crs_type(x, y, coords_list[3]))
+      end
+    end
+  catch e
+    # If CRS construction fails, fall back to raw coordinates
+    @warn "Failed to create point with CRS $crs_type: $e. Using raw coordinates."
+    return Point(coords_list...)
+  end
+end
+
+function parse_wkb_point(io::IOBuffer, byte_order::UInt8, has_z::Bool, has_m::Bool, crs_type)
+  coords_list = read_coordinate_values(io, byte_order, has_z, has_m)
+  return create_point_from_coords(coords_list, crs_type)
+end
+
+function parse_wkb_linestring(io::IOBuffer, byte_order::UInt8, has_z::Bool, has_m::Bool, crs_type; as_rope::Bool=false)
+  read_value = byte_order == 0x01 ? ltoh : ntoh
+  
+  num_points = read_value(read(io, UInt32))
+  
+  if num_points == 0
+    return nothing
+  end
+  
+  # Parse all points using helper functions
+  points = Point[]
+  
+  for i in 1:num_points
+    coords_list = read_coordinate_values(io, byte_order, has_z, has_m)
+    point = create_point_from_coords(coords_list, crs_type)
+    if !isnothing(point)
+      push!(points, point)
+    end
+  end
```

Comment:
Julia offers a more idiomatic method to create a list of objects without explicit type initialization. Instead of writing

```jl
points = Point[]
for i in 1:n
  push!(list, point(i))
end
```

you can write

```jl
points = map(1:n) do i
  point(i)
end
```

The latter method will return a vector of points with concrete type. The type `Point` is under-specified (abstract) and that kills performance.